src/lib/profiles/data-management/Current/TraitData.cpp

```diff
@@ -88,6 +88,66 @@ WEAVE_ERROR UpdateDictionaryDirtyPathCut::CutPath (PropertyPathHandle aPathhandl
     return err;
 }
 
+WEAVE_ERROR TraitSchemaEngine::MapPathToHandle(const char * aPathString, PropertyPathHandle & aHandle) const
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    PropertyPathHandle childProperty, curProperty;
+    nl::Weave::TLV::TLVType dummyContainerType = kTLVType_Path;
+    char *token = NULL;
+    char *pathStringCopy = (char *)malloc(strlen(aPathString) + 1);
+    VerifyOrExit(pathStringCopy != NULL, err = WEAVE_ERROR_NO_MEMORY);
+
+    strcpy(pathStringCopy, aPathString);
+    token = strtok(pathStringCopy, "/");
```

Comment:
strtok is not thread-safe, so it cannot be used here.

Reply:
Please verify that all string-form paths start with /.

Reply:
thanks for your tips, yep, strtok is not thread-safe, we cannot use it here. just update the PR, I have written ParseTagString function to manually parse string-form path start with /, thanks